Easydict/Feature/ViewController/View/Titlebar/EZTitlebar.m

```diff
@@ -199,6 +219,42 @@ - (void)updatePinButtonImage {
     }];
 }
 
+- (void)showMenu {
+    NSMenu * menu = [[NSMenu alloc]initWithTitle:@"Menu"];
+    NSMenuItem * item1 = [[NSMenuItem alloc]initWithTitle:NSLocalizedString(@"automatically_remove_code_comment_symbols", nil) action:@selector(clickAutomaticallyRemoveCodeCommentSymbols) keyEquivalent:@""];
+    item1.target = self;
+    if (Configuration.shared.automaticallyRemoveCodeCommentSymbols) {
+        item1.state = NSControlStateValueOn;
+    }
+    
+    NSMenuItem * item2 = [[NSMenuItem alloc]initWithTitle:NSLocalizedString(@"automatic_word_segmentation", nil) action:@selector(clickAutomaticWordSegmentation) keyEquivalent:@""];
+    item2.target = self;
+    if (Configuration.shared.automaticWordSegmentation) {
+        item2.state = NSControlStateValueOn;
+    }
+    
+    NSMenuItem * item3 = [[NSMenuItem alloc]initWithTitle:NSLocalizedString(@"go_to_settings", nil) action:@selector(goToSettings) keyEquivalent:@""];
+    item3.target = self;
+    
+    [menu addItem:item1];
+    [menu addItem:item2];
+    [menu addItem:[NSMenuItem separatorItem]];
+    [menu addItem:item3];
+    
+    [menu popUpMenuPositioningItem:nil atLocation:[NSEvent mouseLocation] inView:nil];
+}
+
+- (void)clickAutomaticallyRemoveCodeCommentSymbols {
+    Configuration.shared.automaticallyRemoveCodeCommentSymbols = !Configuration.shared.automaticallyRemoveCodeCommentSymbols;
+}
+
+- (void)clickAutomaticWordSegmentation {
+    Configuration.shared.automaticWordSegmentation = !Configuration.shared.automaticWordSegmentation;
+}
+
+- (void)goToSettings {
+    [EZPreferencesWindowController.shared show];
```

Comment:
Why open the old settings when clicking `Go to Settings`?

<img width="899" alt="image" src="https://github.com/tisfeng/Easydict/assets/25194972/a990d29b-650c-4be6-a42d-3c21de4278fe">

Reply:
It doesn't seem to be able to open settings programmatically in MacOS 14.

Reply:
I am not very familiar with SwiftUI Settings, and I'm not sure if there is a convenient solution here.

If there is no better solution, I have a simple and rude idea here, which is to open the settings by simulating the shortcut `Cmd + ,`, like this

```objc
[EZSystemUtility postKeyboardEvent:kCGEventFlagMaskCommand virtualKey:kVK_ANSI_Comma keyDown:true];
```

Reply:
I made some changes in my branch(https://github.com/phlpsong/Easydict/tree/open_settings_appkit) to achieve this. I can create a PR later if this workaround is acceptable.

Reply:
No, it's not good, open url `easydictd://settings` will query word "settings" in Easydict rather than open settings.

Also, using the URL Scheme method is not convenient for debugging because we are not sure if it will open the debug or release Easydict App.

Reply:
URL schema does not works well after we register the handler for AppleEvent.  Currently, I am using notification to handle this. You could check the latest code.

Reply:
Seems no bad, it works as expected although a bit complicated.

Please submit a PR, let's review the code carefully.

Reply:
Yes, indeed.

PR created: https://github.com/NeverAgain11/Easydict/pull/1

Reply:
PR was merged